src/stack.jl

```diff
@@ -258,10 +258,10 @@ function RasterStack(filename::AbstractString;
         end
         RasterStack(joinpath.(Ref(filename), filenames); keys)
     else
-        st = if haslayers(_sourcetype(filename))
+        st = if haslayers(source)
             crs = defaultcrs(source, crs)
             mappedcrs = defaultmappedcrs(source, mappedcrs)
-            data, field_kw = _open(filename) do ds
+            data, field_kw = _open(filename, source=source) do ds
```

Comment:
```suggestion
            data, field_kw = _open(filename; source) do ds
```

Since I think v1.6 we can just use the keyword directly. Also better to always use `;` rather than `,` to separate keywords.